Add quicksnip extension

lille-morille opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

A new tool called [QuickSnip](https://github.com/dostonnabotov/quicksnip/tree/main) created by [Technophile on YoutTube](https://www.youtube.com/watch?v=BhRi7fJzPgk) was launched recently.

@minhnhatnguyen8200 proposed this as a candiate for a Raycast extension, to quickly grab code snippets and copy them to the clipboard.

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->


https://github.com/user-attachments/assets/d28f7754-a0f4-409b-9d3d-fc0a87dc62be

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

- I don't think the icon works light mode, maybe you can use https://icon.ray.so to help with the logo?

- Why only allow copy and not paste?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

lille-morille commented:
Hey!
Thanks for the feedback. I have addressed the issues you described.
However, I am unable to pull your contributions into my repository. I'm getting this error message:
```
❯ npm run publish                                  

> publish
> npx @raycast/api@latest publish

(node:1067) ExperimentalWarning: CommonJS module /opt/homebrew/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /opt/homebrew/lib/node_modules/npm/node_modules/supports-color/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
wait  - validate package.json file ..  /Users/mori/dev/personal/quicksnip/package.json
  4:11  warning  Extension's title has to be Title Cased. Expected "Quicksnip"
ready  - validate package.json file
ready  - validate package-lock.json
ready  - validate other lock files
ready  - validate extension icons
ready  - validate extension metadata
ready  - run ESLint
ready  - run Prettier 3.4.2
ready  - getting fork
ready  - preparing clone
error  - checking for new contributions
some contributions are available. Pull them using `npx @raycast/api@latest pull-contributions`

quicksnip on  main via  v23.3.0 took 10s 
❯ npx @raycast/api@latest pull-contributions
(node:1338) ExperimentalWarning: CommonJS module /opt/homebrew/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /opt/homebrew/lib/node_modules/npm/node_modules/supports-color/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
ready  - getting fork
ready  - preparing clone
error  - pulling new contributions
failed running git Command failed: git pull origin ext/quicksnip --ff-only
From https://github.com/lille-morille/raycast-extensions
 * branch                ext/quicksnip -> FETCH_HEAD
fatal: Not possible to fast-forward, aborting.


From https://github.com/lille-morille/raycast-extensions
 * branch                ext/quicksnip -> FETCH_HEAD
fatal: Not possible to fast-forward, aborting.
```

It seems the `pull-contributions` command does not support merging/rebasing to fix the merge conflicts? Or am I missing something?

pernielsentikaer commented:
Let me try to close this PR and you can delete `lille-morille:ext/quicksnip` and then we can try again, does that sound like a plan?

lille-morille commented:
Created a new pull request [here](https://github.com/raycast/extensions/pull/16500).
I manually added your changes as a commit to the new pull request.
